packages/repository/src/repository-mixin.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { Class } from './common-types';
+import { MixinBuilder } from './mixin';
+import { Component } from '@loopback/core';
+import { Repository } from './repository';
+
+// tslint:disable:no-any
+
+/**
+* A mixin class for Application that creates a .repository()
+* function to register a repository automatically. Also overrides
+* component function to allow it to register repositories automatically.
+*
+* ```ts
+*
+* class MyApplication extends RepositoryMixin(Application) {}
+* ```
+*/
+export function RepositoryMixin<T extends Class<any>>(superClass: T) {
```

Comment:
`T extends Class<Application>`?

Reply:
If I change `any` to `Application` I get the following errors:

```
loopback-next$ tsc -p packages/repository/tsconfig.build.json --target es2017 --outDir packages/repository/lib
packages/repository/src/repository-mixin.ts(22,49): error TS2304: Cannot find name 'Application'.
packages/repository/src/repository-mixin.ts(22,49): error TS4016: Type parameter 'T' of exported function has or is using private name 'Application'.
packages/repository/src/repository-mixin.ts(27,17): error TS2339: Property 'options' does not exist on type '(Anonymous class)'.
packages/repository/src/repository-mixin.ts(27,31): error TS2339: Property 'options' does not exist on type '(Anonymous class)'.
packages/repository/src/repository-mixin.ts(29,16): error TS2339: Property 'options' does not exist on type '(Anonymous class)'.
packages/repository/src/repository-mixin.ts(30,33): error TS2339: Property 'options' does not exist on type '(Anonymous class)'.
packages/repository/src/repository-mixin.ts(35,16): error TS2339: Property 'options' does not exist on type '(Anonymous class)'.
packages/repository/src/repository-mixin.ts(37,38): error TS2339: Property 'options' does not exist on type '(Anonymous class)'.
packages/repository/src/repository-mixin.ts(72,12): error TS2339: Property 'bind' does not exist on type '(Anonymous class)'.
packages/repository/src/repository-mixin.ts(109,33): error TS2339: Property 'getSync' does not exist on type '(Anonymous class)'.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @loopback/repository@4.0.0-alpha.6 build:current: `node ../../bin/compile-package`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @loopback/repository@4.0.0-alpha.6 build:current script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/taranveer/.npm/_logs/2017-09-01T19_55_11_246Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @loopback/repository@4.0.0-alpha.6 pretest: `npm run build:current`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @loopback/repository@4.0.0-alpha.6 pretest script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/taranveer/.npm/_logs/2017-09-01T19_55_11_267Z-debug.log
```